src/Numerics/ODESolvers/BackwardEulerSolvers.jl

```diff
@@ -128,9 +172,147 @@ function update_backward_Euler_solver!(lin::LinBESolver, Q, α)
 end
 
 function (lin::LinBESolver)(Q, Qhat, α, p, t)
+    rhs! = EulerOperator(lin.f_imp!, -α)
+
     if lin.α != α
         @assert lin.isadjustable
         update_backward_Euler_solver!(lin, Q, α)
     end
-    linearsolve!(lin.factors, lin.solver, Q, Qhat, p, t)
+
+    if typeof(lin.solver) <: AbstractIterativeSystemSolver
+        FT = eltype(α)
+        preconditioner_update!(
+            rhs!,
+            rhs!.f!,
+            lin.preconditioner,
+            nothing,
+            FT(NaN),
+        )
+        linearsolve!(rhs!, lin.preconditioner, lin.solver, Q, Qhat, p, t)
+        preconditioner_counter_update!(lin.preconditioner)
+    else
+        linearsolve!(rhs!, lin.factors, lin.solver, Q, Qhat, p, t)
+    end
+end
+
+"""
+struct NonLinearBackwardEulerSolver{NLS}
+    nlsolver::NLS
+    isadjustable::Bool
+    preconditioner_update_freq::Int64
+end
```

Comment:
```suggestion
    struct NonLinearBackwardEulerSolver{NLS}
        nlsolver::NLS
        isadjustable::Bool
        preconditioner_update_freq::Int64
    end
```
Documenter treats indented lines (with 4 spaces) specially.